com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/CompletionProposalRequestor.java

```diff
@@ -160,9 +160,16 @@ public CompletionItem toCompletionItem(CompletionProposal proposal, int index) {
         data.put(CompletionResolveHandler.DATA_FIELD_REQUEST_ID, String.valueOf(response.getId()));
         data.put(CompletionResolveHandler.DATA_FIELD_PROPOSAL_ID, String.valueOf(index));
         $.setData(data);
+
         this.descriptionProvider.updateDescription(proposal, $);
         // Use fully qualified name as needed.
-        $.setInsertText(String.valueOf(proposal.getCompletion()));
+        String insertText = String.valueOf(proposal.getCompletion());
+        int prefix = proposal.getReplaceEnd() - proposal.getReplaceStart();
+        if (prefix > 0) {
+          $.setInsertText(insertText.substring(prefix));
+        } else {
+          $.setInsertText(insertText);
+        }
```

Comment:
this doesn't work for VS Code. Typing `Sys` to trigger completion, select `System` and it becomes `tem`.